pkg/wallet/internal/mapping/mapping_list_actions.go

```
@@ -0,0 +1,187 @@
+	}
+
+	if args.IncludeOutputLockingScripts != nil {
+		includeOutputLockingScripts := primitives.BooleanDefaultFalse(*args.IncludeOutputLockingScripts)
+		result.IncludeOutputLockingScripts = &includeOutputLockingScripts
+	}
+
+	return result
+}
+
+// MapListActionsResult maps *wdk.ListActionsResult to *sdk.ListActionsResult
+func MapListActionsResult(result *wdk.ListActionsResult) *sdk.ListActionsResult {
+	return &sdk.ListActionsResult{
+		TotalActions: result.TotalActions,
+		Actions:      slices.Map(result.Actions, mapListActionsAction),
+	}
+}
+
+// mapListActionsAction maps wdk.WalletAction to sdk.Action
+func mapListActionsAction(action wdk.WalletAction) sdk.Action {
+	var txidHash chainhash.Hash
+	if action.TxID != "" {
+		if hash, err := chainhash.NewHashFromHex(action.TxID); err == nil {
+			txidHash = *hash
+		}
+	}
+
+	if action.Satoshis < 0 {
+		panic("negative satoshis value encountered: data integrity issue")
+	}
+
+	result := sdk.Action{
+		Txid:        txidHash,
+		Satoshis:    uint64(action.Satoshis),
+		Status:      mapActionStatus(action.Status),
+		IsOutgoing:  action.IsOutgoing,
+		Description: action.Description,
+		Labels:      action.Labels,
+		Version:     action.Version,
+		LockTime:    action.LockTime,
+		Inputs:      slices.Map(action.Inputs, mapActionInput),
+		Outputs:     slices.Map(action.Outputs, mapActionOutput),
+	}
+
+	return result
+}
+
+// mapActionStatus maps string status to sdk.ActionStatus
+func mapActionStatus(status string) sdk.ActionStatus {
+	switch status {
+	case "completed":
+		return sdk.ActionStatusCompleted
+	case "unprocessed":
+		return sdk.ActionStatusUnprocessed
+	case "sending":
+		return sdk.ActionStatusSending
+	case "unproven":
+		return sdk.ActionStatusUnproven
+	case "unsigned":
+		return sdk.ActionStatusUnsigned
+	case "nosend":
+		return sdk.ActionStatusNoSend
+	case "nonfinal":
+		return sdk.ActionStatusNonFinal
+	default:
+		return sdk.ActionStatusUnprocessed
```

shouldn't we return an error here, that unknown status was received.